api/apps/geoprocessing/src/modules/planning-units-grid/planning-units-grid.processor.ts

```diff
@@ -40,27 +40,77 @@ export class PlanningUnitsGridProcessor
 
     const result = this.gridGeoJsonValidator.validate(geoJson);
     if (isLeft(result)) {
-      throw new Error(result.left.toString());
+      const error = new Error(result.left.toString());
+      this.logger.error(error);
+      throw error;
     }
-    const puGeometriesIds: { id: string }[] = await this.entityManager.query(
-      `
-        INSERT INTO "planning_units_geom"("the_geom", "type", "project_id")
-        SELECT ST_SetSRID(
-                 ST_GeomFromGeoJSON(features ->> 'geometry'),
-                 4326)::geometry,
-               $2::shape_type,
-               $3
-        FROM (
-               SELECT json_array_elements($1::json -> 'features') AS features
-             ) AS f
-        ON CONFLICT ON CONSTRAINT planning_units_geom_the_geom_type_key DO UPDATE SET type = 'from_shapefile'::shape_type
-        RETURNING "id"
-      `,
-      [result.right, ShapeType.FromShapefile, projectId],
-    );
+
+    const { puGeometriesIds, planningAreaId, bbox } = await this.entityManager
+      .transaction(async (manager) => {
```

Comment:
good point. until there are no scenarios, it's ok for API clients to keep sending a new PU grid, and we'll happily use the latest (with usual caveats about concurrency of rapid-fire identical uploads).

in practice, if a rogue API user hits the PU grid upload endpoint after a scenario has been created, chaos will ensue.

so tl;dr it's ok to delete geoms and pa here, but I think we should add a guard in `ProjectsService.setGrid()` to reject grid uploads if the project has one or more scenarios. If I haven't missed an existing guard and this sounds like a sensible plan to you, it can be done in a separate PR, in any case.